db/post_migrate/20210621234939_backfill_email_log_topic_id.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class BackfillEmailLogTopicId < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+  BATCH_SIZE = 30_000
+
+  def up
+    offset = 0
+    email_log_count = DB.query_single("SELECT COUNT(*) FROM email_logs").first
+
+    loop do
+      DB.exec(<<~SQL, offset: offset, batch_size: BATCH_SIZE)
+          WITH cte AS (
+            SELECT topic_id, post_id
```

Comment:
It seems like we do not have to select `topic_id` here although I'm wondering if we're missing a filter to not update rows `WHERE topic_id IS NOT NULL`

Reply:
Good point, we do not need topic_id here and for the update I will do this:

```
WHERE email_logs.post_id = cte.post_id AND email_logs.topic_id IS NULL
```

Reply:
I think we can add the `email_logs.topic_id IS NULL` query to the CTE instead.

Reply:
Oh right, done

Reply:
O no we can't because the rows may change between queries. haha ignore my comment here.

Reply:
Okay I will remove the additional WHERE lol